src/app/clusters/general-commissioning-server/general-commissioning-server.cpp

```diff
@@ -218,60 +373,90 @@ bool emberAfGeneralCommissioningClusterCommissioningCompleteCallback(
     auto & failSafe               = Server::GetInstance().GetFailSafeContext();
     auto & fabricTable            = Server::GetInstance().GetFabricTable();
 
+    CHIP_ERROR err = CHIP_NO_ERROR;
+
     ChipLogProgress(FailSafe, "GeneralCommissioning: Received CommissioningComplete");
 
     Commands::CommissioningCompleteResponse::Type response;
+
+    // Fail-safe must be armed
     if (!failSafe.IsFailSafeArmed())
     {
         response.errorCode = CommissioningErrorEnum::kNoFailSafe;
+        commandObj->AddResponse(commandPath, response);
+        return true;
     }
-    else
+
+    SessionHandle handle = commandObj->GetExchangeContext()->GetSessionHandle();
+
+    // Ensure it's a valid CASE session
```

Comment:
Per (now updated) spec, this check happens after the terms and conditions check, so this is not matching the spec.

Reply:
I will revert the sequence for now, but IMO we should be validating that the command was received over a valid secure CASE session first, since the spec says the CommissioningComplete command can only be received over a CASE session.

Reply:
I've created an issue for us to clarify the sequence of operations when the CommissioningComplete command is being handled. https://github.com/CHIP-Specifications/connectedhomeip-spec/issues/10684